tests/unit/Rules/SfTest.php

```diff
@@ -14,84 +14,103 @@
 namespace Respect\Validation\Rules;
 
 use PHPUnit\Framework\TestCase;
-use Respect\Validation\Exceptions\AllOfException;
-use Respect\Validation\Validator as v;
+use Respect\Validation\Exceptions\SfException;
+use Symfony\Component\Validator\Constraints\Collection;
+use Symfony\Component\Validator\Constraints\IsFalse;
+use Symfony\Component\Validator\Constraints\IsNull;
+use Symfony\Component\Validator\Validation;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use const PHP_EOL;
+use function implode;
 
 /**
- * @group  rule
- * @covers \Respect\Validation\Exceptions\SfException
+ * @group rule
+ *
  * @covers \Respect\Validation\Rules\Sf
+ *
+ * @author Augusto Pascutti <augusto@phpsp.org.br>
+ * @author Gabriel Caruso <carusogabriel34@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
  */
-class SfTest extends TestCase
+final class SfTest extends TestCase
 {
     /**
      * @test
      */
-    public function validationWithAnExistingValidationConstraint(): void
+    public function itShouldValidateWithDefinedConstraintAndValidator(): void
     {
-        $constraintName = 'Time';
-        $validConstraintValue = '04:20:00';
-        $invalidConstraintValue = 'yada';
-        self::assertTrue(
-            v::sf($constraintName)->validate($validConstraintValue),
-            sprintf('"%s" should be valid under "%s" constraint.', $validConstraintValue, $constraintName)
-        );
-        self::assertFalse(
-            v::sf($constraintName)->validate($invalidConstraintValue),
-            sprintf('"%s" should be invalid under "%s" constraint.', $invalidConstraintValue, $constraintName)
-        );
+        $constraint = new IsNull();
+        $validator = Validation::createValidator();
+
+        $sut = new Sf($constraint, $validator);
+
+        self::assertTrue($sut->validate(null));
```

Comment:
how can we assert that the Validator provided is a custom one? here it only validates the result.